From 9c32ff964279df67d4fdac583f3ad6a97cead1b0 Mon Sep 17 00:00:00 2001
From: Sanjeev Jakkilinki <sanjeevj@codeaurora.org>
Date: Tue, 25 Apr 2017 16:17:54 +0530
Subject: [PATCH] frameworks/base: reduce the setTransactions

reduce the setTransactions from window manager service to
reduce the janks during animations for more numbered layers

Test: Build successful, visible performance improvement.

diff --git a/services/core/java/com/android/server/wm/DimLayer.java b/services/core/java/com/android/server/wm/DimLayer.java
index 052b2f5..4e00176 100644
--- a/services/core/java/com/android/server/wm/DimLayer.java
+++ b/services/core/java/com/android/server/wm/DimLayer.java
@@ -201,13 +201,14 @@
         }
 
         if (mDimSurface != null) {
-            mDimSurface.setPosition(mBounds.left, mBounds.top);
-            mDimSurface.setSize(mBounds.width(), mBounds.height());
-            if (DEBUG_DIM_LAYER) Slog.v(TAG,
-                    "adjustBounds user=" + mUser.toShortString() + " mBounds=" + mBounds);
+            if(!mLastBounds.equals(mBounds)) {
+                 mDimSurface.setPosition(mBounds.left, mBounds.top);
+                 mDimSurface.setSize(mBounds.width(), mBounds.height());
+                 if (DEBUG_DIM_LAYER) Slog.v(TAG,
+                        "adjustBounds user=" + mUser.toShortString() + " mBounds=" + mBounds);
+                 mLastBounds.set(mBounds);
+           }
         }
-
-        mLastBounds.set(mBounds);
     }
 
     private void getBoundsForFullscreen(Rect outBounds) {
